core/arch/arm/kernel/secure_partition.c

```diff
@@ -421,32 +600,52 @@ static TEE_Result sp_init_uuid(const TEE_UUID *uuid, const void * const fdt)
 	vaddr_t va = 0;
 	size_t num_pgs = 0;
 	struct sp_ctx *ctx = NULL;
+	void *fdt_copy = NULL;
+	const char *dt_memory_match_table = {
+		"arm,ffa-manifest-memory-regions",
+	};
+	const char *dt_device_match_table = {
+		"arm,ffa-manifest-device-regions",
+	};
 
-	res = handle_fdt(fdt, uuid);
-
-	if (res)
-		return res;
 
 	res = sp_open_session(&sess,
 			      &open_sp_sessions,
 			      uuid);
 	if (res)
 		return res;
 
+	res = handle_fdt(fdt, uuid);
+
+	if (res)
+		return res;
+
 	ctx = to_sp_ctx(sess->ts_sess.ctx);
 	ts_push_current_session(&sess->ts_sess);
-	res = sp_init_info(ctx, &args, fdt, &va, &num_pgs);
-	ts_pop_current_session();
+	res = sp_init_info(ctx, &args, fdt, &va, &num_pgs, &fdt_copy);
 	if (res)
-		return res;
+		goto out;
 
-	if (sp_enter(&args, sess))
+	res = handle_fdt_regions(fdt_copy, sess, dt_device_match_table, true);
+	if (res)
+		goto out;
+
+	res = handle_fdt_regions(fdt_copy, sess, dt_memory_match_table, false);
+	if (res)
+		goto out;
+
+	ts_pop_current_session();
+
+	if (sp_enter(&args, sess)) {
+		vm_unmap(&ctx->uctx, va, num_pgs);
 		return FFA_ABORTED;
+	}
 
 	spmc_sp_msg_handler(&args, sess);
 
-	/* Free the boot info page from the SP memory.*/
 	ts_push_current_session(&sess->ts_sess);
+out:
+	/* Free the boot info page from the SP memory.*/
 	res = vm_unmap(&ctx->uctx, va, num_pgs);
```

Comment:
This line shadows the previous `res` value so the failure of `handle_fdt_regions` won't propagate to the caller.

Reply:
Fixed